m2cgen/exporters.py

```diff
@@ -19,92 +24,108 @@ def export_to_java(model, package_name=None, class_name="Model", indent=4):
+    output_file : string, optional
+        Path to a file in which the generated code should be written.
     indent : int, optional
         The size of indents in the generated code.
 
     Returns
     -------
-    code : string
+    code : string or None
     """
     interpreter = interpreters.PythonInterpreter(indent=indent)
-    return _export(model, interpreter)
+    return _export(model, interpreter, output_file)
 
 
-def export_to_c(model, indent=4):
+def export_to_c(model, output_file=None, indent=4):
     """
     Generates a C code representation of the given model.
 
     Parameters
     ----------
     model : object
         The model object that should be transpiled into code.
+    output_file : string, optional
+        Path to a file in which the generated code should be written.
     indent : int, optional
         The size of indents in the generated code.
 
     Returns
     -------
-    code : string
+    code : string or None
     """
     interpreter = interpreters.CInterpreter(indent=indent)
-    return _export(model, interpreter)
+    return _export(model, interpreter, output_file)
 
 
-def export_to_go(model, indent=4):
+def export_to_go(model, output_file=None, indent=4):
     """
     Generates a Go code representation of the given model.
 
     Parameters
     ----------
     model : object
         The model object that should be transpiled into code.
+    output_file : string, optional
+        Path to a file in which the generated code should be written.
     indent : int, optional
         The size of indents in the generated code.
 
     Returns
     -------
-    code : string
+    code : string or None
     """
     interpreter = interpreters.GoInterpreter(indent=indent)
-    return _export(model, interpreter)
+    return _export(model, interpreter, output_file)
 
 
-def export_to_javascript(model, indent=4):
+def export_to_javascript(model, output_file=None, indent=4):
     """
     Generates a Javascript code representation of the given model.
 
     Parameters
     ----------
     model : object
         The model object that should be transpiled into code.
+    output_file : string, optional
+        Path to a file in which the generated code should be written.
     indent : int, optional
         The size of indents in the generated code.
 
     Returns
     -------
-    code : string
+    code : string or None
     """
     interpreter = interpreters.JavascriptInterpreter(indent=indent)
-    return _export(model, interpreter)
+    return _export(model, interpreter, output_file)
 
 
-def _export(model, interpreter):
+def _export(model, interpreter, output_file=None):
     assembler_cls = assemblers.get_assembler_cls(model)
     model_ast = assembler_cls(model).assemble()
-    return interpreter.interpret(model_ast)
+    code = interpreter.interpret(model_ast)
+    if output_file is None:
+        return code
+    else:
+        directory = os.path.dirname(output_file)
```

Comment:
I'd say we should not only support string file names but file descriptors (`write()` method) as well.

Reply:
I'd also add that I'm not sure that `m2cgen` should implicitly create directories.